src/modules/RA/Scene/components/GeolayerSelect.js

```diff
@@ -91,11 +135,23 @@ const GeolayerSelect = ({ dataProvider, onChange, excludeIds = [], includeIds =
   return (
     <FormControl fullWidth>
       <InputLabel htmlFor="geolayer">Couche</InputLabel>
-      <Select onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
-        {geolayers.map(({ id, name }) => (
-          <MenuItem key={id} value={id}>{name}</MenuItem>
-        ))}
-      </Select>
+      { (geolayers.length > MAX_NON_NATIVE_SELECT)
+        // Do we have too many element to have perf issues ?
+        ? (
+          <NativeSelect onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
+            {geolayers.map(({ id, name }) => (
+              <option key={id} value={id}>{name}({id})</option>
+            ))}
+          </NativeSelect>
+        )
+        : (
+          <Select onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
+            {geolayers.map(({ id, name }) => (
+              <MenuItem key={id} value={id}>{name}({id})</MenuItem>
+            ))}
+          </Select>
+        )
+      }
```

Comment:
Always prefer avoiding ternary operator for multiline JSX:

```suggestion
      {/* Do we have too many element to have perf issues ? */}
      {(geolayers.length > MAX_NON_NATIVE_SELECT) && (
          <NativeSelect onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
            {geolayers.map(({ id, name }) => (
              <option key={id} value={id}>{name}({id})</option>
            ))}
          </NativeSelect>
        )}

        {(geolayers.length <= MAX_NON_NATIVE_SELECT) && (
          <Select onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
            {geolayers.map(({ id, name }) => (
              <MenuItem key={id} value={id}>{name}({id})</MenuItem>
            ))}
          </Select>
        )}
```

Reply:
Done